lib/adapters/http/http.js

```diff
@@ -73,7 +73,7 @@ function getHost(name, opts) {
     uri.path = parts.join('/');
     opts = opts || {};
     opts = utils.clone(opts);
-    uri.headers = opts.headers || {};
+    uri.headers = opts.headers || (opts.ajax && opts.ajax.headers) || {};
 
     if (opts.auth || uri.auth) {
       var nAuth = opts.auth || uri.auth;
```

Comment:
Heres the test - https://github.com/pouchdb/pouchdb/pull/3418/files#diff-e71e1c5c6d0a286d583590ac933b89eeR35